Lib/test/test_sysconfig.py

```diff
@@ -592,6 +593,54 @@ def test_osx_ext_suffix(self):
         suffix = sysconfig.get_config_var('EXT_SUFFIX')
         self.assertTrue(suffix.endswith('-darwin.so'), suffix)
 
+    def test_always_set_abiflags(self):
+        self.assertIn('ABIFLAGS', sysconfig.get_config_vars())
+        self.assertIsInstance(sysconfig.get_config_var('ABIFLAGS'), str)
+        self.assertIn('abiflags', sysconfig.get_config_vars())
+        self.assertIsInstance(sysconfig.get_config_var('abiflags'), str)
+
+    def test_always_set_py_debug(self):
+        self.assertIn('Py_DEBUG', sysconfig.get_config_vars())
+        Py_DEBUG = sysconfig.get_config_var('Py_DEBUG')
+        self.assertIn(Py_DEBUG, (0, 1))
+        self.assertEqual(Py_DEBUG, support.Py_DEBUG)
+
+    def test_always_set_py_gil_disabled(self):
+        self.assertIn('Py_GIL_DISABLED', sysconfig.get_config_vars())
+        Py_GIL_DISABLED = sysconfig.get_config_var('Py_GIL_DISABLED')
+        self.assertIn(Py_GIL_DISABLED, (0, 1))
+        self.assertEqual(Py_GIL_DISABLED, support.Py_GIL_DISABLED)
+
+    def test_abiflags(self):
```

Comment:
This entire test is unnecessary. Testing that `sysconfig.get_config_var('ABIFLAGS')` is not `None` is sufficient (it's allowed to be empty, and the only definition for it is "what value did it have when we built").

Reply:
> This entire test is unnecessary.

It is still worth testing if `d` is the last flag present.

> A better test (for Windows) would be that _winapi.GetModuleFileName(sys.dllhandle) == f"python{sys.version_info.major}{sys.version_info.minor}{sysconfig.get_config_var('ABIFLAGS')}.dll

Added.

Reply:
If you're going to insist, then also add a comment to the test saying that ABIFLAGS has no definition, and if the ABIFLAGS for a particular build doesn't pass the test, assume the test should be fixed.